device/vr/android/tango/tango_vr_device.cc

```diff
@@ -213,41 +214,58 @@ std::vector<mojom::VRHitPtr> TangoVRDevice::HitTest(float x, float y)
   return mojomHits;
 }
 
-std::vector<mojom::VRPlanePtr> TangoVRDevice::GetPlanes() {
-  std::vector<mojom::VRPlanePtr> mojomPlanes;
-  if (TangoHandler::getInstance()->isConnected())
+mojom::VRPlanePtr CreateMojomPlane(Plane& plane) {
+  mojom::VRPlanePtr result = mojom::VRPlane::New();
+
+  result->identifier = plane.identifier;
+
+  result->modelMatrix.resize(16);
+  for (int i = 0; i < 16; i++)
   {
-    std::vector<Plane> planes;
-    if (TangoHandler::getInstance()->getPlanes(planes) && planes.size() > 0)
-    {
-      std::vector<Plane>::size_type size = planes.size();
-      mojomPlanes.resize(size);
-      for (std::vector<Plane>::size_type i = 0; i < size; i++)
-      {
-        mojomPlanes[i] = mojom::VRPlane::New();
+    result->modelMatrix[i] = plane.modelMatrix[i];
+  }
 
-        mojomPlanes[i]->identifier = planes[i].identifier;
+  result->extent.resize(2);
+  for (int i = 0; i < 2; i++) {
+    result->extent[i] = plane.extent[i];
+  }
 
-        mojomPlanes[i]->modelMatrix.resize(16);
-        for (int j = 0; j < 16; j++)
-        {
-          mojomPlanes[i]->modelMatrix[j] = planes[i].modelMatrix[j];
-        }
+  result->vertices.resize(plane.count * 3);
+  for (uint i = 0; i < plane.count * 3; i++) {
+    result->vertices[i] = plane.vertices[i];
+  }
+  result->count = plane.count;
 
-        mojomPlanes[i]->extent.resize(2);
-        for (int j = 0; j < 2; j++) {
-          mojomPlanes[i]->extent[j] = planes[i].extent[j];
-        }
+  return result;
+}
 
-        mojomPlanes[i]->vertices.resize(planes[i].count * 3);
-        for (uint j = 0; j < planes[i].count * 3; j++) {
-          mojomPlanes[i]->vertices[j] = planes[i].vertices[j];
-        }
-        mojomPlanes[i]->count = planes[i].count;
+void PopulateMojomPlanes(std::vector<mojom::VRPlanePtr>& mojomPlanes, 
```

Comment:
These method look like they could be static.

Reply:
Done